tests/e2e/vm_label_annotation_test.go

```diff
@@ -18,263 +18,295 @@ package e2e
+
+			It(fmt.Sprintf("removes annotation %s from VMs", specialKeyValue), func() {
+				res := kubectl.List(kc.ResourceVM, kc.GetOptions{
+					Labels:    testCaseLabel,
+					Namespace: conf.Namespace,
+					Output:    "jsonpath='{.items[*].metadata.name}'",
+				})
+				Expect(res.Error()).NotTo(HaveOccurred(), "cmd: %s\nstderr: %s", res.GetCmd(), res.StdErr())
+
+				vms := strings.Split(res.StdOut(), " ")
+				err := RemoveAnnotation(kc.ResourceVM, specialKeyValue, vms...)
+				Expect(err).NotTo(HaveOccurred(), err)
 			})
-		})
 
-		Describe("Remove annotation test-annotation=true", func() {
-			It("Was removed", func() {
-				subCMD := fmt.Sprintf("-n %s annotate vm %s %s-", conf.Namespace, vm.Name, annotationName)
-				res := kubectl.RawCommand(subCMD, ShortWaitDuration)
-				Expect(res.Error()).NotTo(HaveOccurred())
+			It("checks VMs and pods annotations after VMs unannotating", func() {
+				res := kubectl.List(kc.ResourceVM, kc.GetOptions{
+					Labels:    testCaseLabel,
+					Namespace: conf.Namespace,
+					Output:    "jsonpath='{.items[*].metadata.name}'",
+				})
+				Expect(res.Error()).NotTo(HaveOccurred(), "cmd: %s\nstderr: %s", res.GetCmd(), res.StdErr())
+
+				vms := strings.Split(res.StdOut(), " ")
+				for _, vm := range vms {
+					vmObj := virtv2.VirtualMachine{}
+					err := GetObject(kc.ResourceVM, vm, &vmObj, kc.GetOptions{Namespace: conf.Namespace})
+					Expect(err).NotTo(HaveOccurred(), err)
+					Expect(vmObj.Annotations).ShouldNot(HaveKey("specialKey"))
+
+					activePod := GetActiveVirtualMachinePod(&vmObj)
+					vmPodObj := v1.Pod{}
+					err = GetObject(kc.ResourcePod, activePod, &vmPodObj, kc.GetOptions{Namespace: conf.Namespace})
+					Expect(err).NotTo(HaveOccurred(), err)
+					Expect(vmPodObj.Annotations).ShouldNot(HaveKey("specialKey"))
+				}
 			})
 		})
+	})
 
-		Describe("Annotation must be removed from resource", func() {
-			It("VM", func() {
-				res := getRecourseAnnotation(kc.ResourceVM, vm.Name)
-				Expect(res.Error()).NotTo(HaveOccurred(), "failed to get VM %s.\n%s", vm.Name, res.StdErr())
-				Expect(res.StdOut()).To(BeEmpty())
-			})
-			It("KVVM", func() {
-				res := getRecourseAnnotation(kc.ResourceKubevirtVM, vm.Name)
-				Expect(res.Error()).NotTo(HaveOccurred(), "failed to get KVVM %s.\n%s", vm.Name, res.StdErr())
-				Expect(res.StdOut()).To(BeEmpty())
-			})
-			It("KVVMI", func() {
-				res := getRecourseAnnotation(kc.ResourceKubevirtVMI, vm.Name)
-				Expect(res.Error()).NotTo(HaveOccurred(), "failed to get KVVMI %s.\n%s", vm.Name, res.StdErr())
-				Expect(res.StdOut()).To(BeEmpty())
-			})
-			It("POD virtlauncher", func() {
-				pod := getPodName(vm.Name)
-				res := getRecourseAnnotation(kc.ResourcePod, pod)
-				Expect(res.Error()).NotTo(HaveOccurred(), "failed to get pod %s.\n%s", vm.Name, res.StdErr())
-				Expect(res.StdOut()).To(BeEmpty())
+	Context("When test is complited:", func() {
+		It("tries to delete used resources", func() {
```

Comment:
Also,  cleanup operation seems strange to be a part of the test suite. Shouldn't it belongs to the test setup actions, something like AfterAll?

Reply:
I put it to wrapper function with description. We need this context to check that all resources in test cases can be deleted correctly.